homeassistant/components/esphome/entry_data.py

```diff
@@ -125,18 +133,65 @@ async def async_update_static_infos(
         async_dispatcher_send(hass, signal, infos)
 
     @callback
-    def async_update_state(self, hass: HomeAssistant, state: EntityState) -> None:
+    def register_state_type(
```

Comment:
Do we really need this to be so dynamic or could we create a static dict mapping state classes to the `Platforms` enum ?

Reply:
I'll change it to a dict

Reply:
That makes everything a lot simpler and in-line with the existing `INFO_TYPE_TO_PLATFORM` mapping